scene/2d/joint_2d.cpp

```diff
@@ -435,6 +435,8 @@ void GrooveJoint2D::_configure_joint(RID p_joint, PhysicsBody2D *body_a, Physics
 
 void GrooveJoint2D::set_length(real_t p_length) {
 	length = p_length;
+	_disconnect_signals();
```

Comment:
You should add this to `set_initial_offset` as well

Reply:
Yes, also seems other joint2d are fine.